homeassistant/components/yamaha_musiccast/config_flow.py

```diff
@@ -0,0 +1,127 @@
+"""Config flow for MusicCast."""
+from __future__ import annotations
+
+import logging
+from urllib.parse import urlparse
+
+from aiohttp import ClientConnectorError
+from aiomusiccast import MusicCastConnectionException, MusicCastDevice
+import voluptuous as vol
+
+from homeassistant.components import ssdp
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_HOST
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.typing import ConfigType
+
+from ... import data_entry_flow
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class MusicCastFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a MusicCast config flow."""
+
+    VERSION = 1
+
+    serial_number: str | None = None
+    model_name: str | None = None
+    host: str
+
+    async def async_step_user(
+        self, user_input: ConfigType | None = None
+    ) -> data_entry_flow.FlowResult:
+        """Handle a flow initiated by the user."""
+        return await self._handle_config_flow(user_input)
+
+    async def _handle_config_flow(
+        self, user_input: ConfigType | None = None
+    ) -> data_entry_flow.FlowResult:
+        """Config flow handler for MusicCast."""
+
+        # Request user input, unless we are preparing discovery flow
+        if user_input is None:
+            return self._show_setup_form()
+
+        title = user_input[CONF_HOST]
+        self.host = user_input[CONF_HOST]
+
+        errors = {}
+        # Check if device is a MusicCast device
+
+        try:
+            info = await MusicCastDevice.get_device_info(
+                user_input[CONF_HOST], async_get_clientsession(self.hass)
+            )
+            if self.serial_number is None:
+                self.serial_number = info["system_id"]
+            unique_id = self.serial_number
+            await self.async_set_unique_id(unique_id)
```

Comment:
We don't want to raise in the user flow if there's a discovery flow in progress. The user flow should take precedence.
```suggestion
            await self.async_set_unique_id(unique_id, raise_on_progress=False)
```